python-checks/src/main/java/org/sonar/python/checks/TooManyParametersCheck.java

```diff
@@ -41,29 +52,71 @@
 
   @Override
   public void initialize(Context context) {
-    context.registerSyntaxNodeConsumer(Kind.FUNCDEF, ctx -> {
-      FunctionDef tree = (FunctionDef) ctx.syntaxNode();
-      if (tree.parameters() != null) {
-        int nbParameters = tree.parameters().all().size();
-        if (nbParameters > max) {
-          String typeName = tree.isMethodDefinition() ? "Method" : "Function";
-          String name = String.format("%s \"%s\"", typeName, tree.name().name());
-          String message = String.format(MESSAGE, name, nbParameters, max);
-          ctx.addIssue(tree.parameters(), message);
-        }
-      }
-    });
+    context.registerSyntaxNodeConsumer(Kind.FUNCDEF, this::checkFunctionDef);
 
     context.registerSyntaxNodeConsumer(Kind.LAMBDA, ctx -> {
       LambdaExpression tree = (LambdaExpression) ctx.syntaxNode();
-      if (tree.parameters() != null) {
-        int nbParameters = tree.parameters().all().size();
+      ParameterList parameters = tree.parameters();
+      if (parameters != null) {
+        int nbParameters = parameters.all().size();
         if (nbParameters > max) {
           String name = "Lambda";
           String message = String.format(MESSAGE, name, nbParameters, max);
-          ctx.addIssue(tree.parameters(), message);
+          ctx.addIssue(parameters, message);
         }
       }
     });
   }
+
+  private void checkFunctionDef(SubscriptionContext ctx) {
+    FunctionDef functionDef = (FunctionDef) ctx.syntaxNode();
+    ParameterList parameters = functionDef.parameters();
+    if (parameters != null) {
+      long nbParameters = parameters.nonTuple().stream().filter(s -> s.starToken() == null || s.name() != null).count();
+      boolean isMethod = functionDef.isMethodDefinition();
+      if (isMethod && functionDef.decorators().stream().noneMatch(d -> d.name().names().stream().anyMatch(n -> n.name().equals("staticmethod")))) {
```

Comment:
I think you could use `FunctionSymbols.decorators()` that is already a list of strings